Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,197 @@
+                guard let tapScreenPoint else { return }
+                // Sets current geoprocessingInProgress to
+                // false after geoprocessing is complete.
+                defer { geoprocessingInProgress = false }
+                model.addGraphic(at: tapScreenPoint)
+                geoprocessingInProgress = true
+                await model.calculateViewshed(at: tapScreenPoint)
+            }
+            .errorAlert(presentingError: $model.error)
+    }
+}
+
+private extension ShowViewshedFromPointOnMapView {
+    class Model: ObservableObject {
+        /// The error shown in the error alert.
+        @Published var error: Error?
+        
+        /// A map with topographic basemap.
+        /// Sets map's initial viewpoint to Vanoise National Park in France.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            map.initialViewpoint = Viewpoint(
+                center: Point(latitude: 45.379, longitude: 6.849),
+                scale: 144447
+            )
+            return map
+        }()
+        
+        /// The processing task that references the online resource through the url.
+        let geoprocessingTask = GeoprocessingTask(url: .viewshedURL)
+        
+        ///  The graphics overlay that displays the viewshed for the tapped location.
+        var resultGraphicsOverlay = {
+            let resultGraphicsOverlay = GraphicsOverlay()
+            let fillColor = UIColor(
+                red: 226 / 255.0,
+                green: 119 / 255.0,
+                blue: 40 / 255,
+                alpha: 120 / 255.0
+            )
+            let fillSymbol = SimpleFillSymbol(
+                style: .solid,
+                color: fillColor,
+                outline: nil
+            )
+            let resultRenderer = SimpleRenderer(symbol: fillSymbol)
+            resultGraphicsOverlay.renderer = resultRenderer
+            return resultGraphicsOverlay
+        }()
+        
+        /// The graphics overlay that shows where the user tapped on the screen.
+        var inputGraphicsOverlay = {
+            let inputGraphicsOverlay = GraphicsOverlay()
+            // Red dot marker that is added to graphic overlay on map on the location where the user taps.
+            let pointSymbol = SimpleMarkerSymbol(style: .circle, color: .red, size: 10)
+            // Sets the renderer to draw the dot symbol on the graphics overlay.
+            let renderer = SimpleRenderer(symbol: pointSymbol)
+            inputGraphicsOverlay.renderer = renderer
+            return inputGraphicsOverlay
+        }()
+        
+        /// Handles the execution of the geoprocessing task and gets the result.
+        var geoprocessingJob: GeoprocessingJob?
+        
+        /// Removes previously tapped location from overlay and draws new dot on tap location.
+        /// - Parameter point: Location that the user tapped on the map.
+        func addGraphic(at point: Point) {
```

Comment:
Rename it to `addInputGraphic` would be more fitting because there are 2 graphics overlays.